File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/models/RoomItem.java
package vng.zalo.tdtai.zalo.zalo.models;

import androidx.annotation.NonNull;

import com.google.firebase.Timestamp;

import java.util.HashMap;
import java.util.Map;

public class RoomItem extends DataModel {
    public String roomId;
    public String avatar;
    public String lastMsg;
    public Timestamp lastMsgTime;
    public String name;
    public int unseenMsgNum;
    public int roomType;

    //mapping firestore data to POJO needs empty constructor
    public RoomItem(){}

    public final Map<String, Object> toMap(){
        Map<String, Object> res = new HashMap<>();

        res.put("avatar",this.avatar);
        res.put("lastMsg",this.lastMsg);
        res.put("lastMsgTime",this.lastMsgTime);
        res.put("name",this.name);
        res.put("unseenMsgNum",this.unseenMsgNum);
        res.put("roomType",this.roomType);

        return res;
    }

    @NonNull
    @Override
    public String toString() {
        return this.toMap().toString();
    }
}

File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/models/RoomItem.kt
package vng.zalo.tdtai.zalo.zalo.models

import com.google.firebase.Timestamp
import vng.zalo.tdtai.zalo.zalo.utils.Constants

import java.util.HashMap

//mapping firestore data to POJO needs empty constructor
data class RoomItem(var roomId: String? = null,
                    var avatar: String? = null,
                    var lastMsg: String? = null,
                    var lastMsgTime: Timestamp? = null,
                    var name: String? = null,
                    var unseenMsgNum: Int = 0,
                    var roomType: Int? = Constants.ROOM_TYPE_PEER,
                    var lastSenderPhone: String? = null) {

    fun toMap(): Map<String, Any?> {
        return HashMap<String, Any?>().apply {
            put("avatar", avatar)
            put("lastMsg", lastMsg)
            put("lastMsgTime", lastMsgTime)
            put("name", name)
            put("unseenMsgNum", unseenMsgNum)
            put("roomType", roomType)
        }
    }
}